diff -pur third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc
--- third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc	2020-09-08 23:38:39.000000000 +0200
+++ third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc	2021-10-07 09:43:35.243632207 +0200
@@ -37,6 +37,7 @@
 
 #include <algorithm>
 #include <array>
+#include <limits>
 #include "absl/base/internal/hide_ptr.h"
 #include "absl/base/internal/raw_logging.h"
 #include "absl/base/internal/spinlock.h"
