diff -pur pytorch-1.8.0-old/c10/core/TensorImpl.h pytorch-1.8.0-old/c10/core/TensorImpl.h
--- pytorch-1.8.0-old/c10/core/TensorImpl.h	2021-10-07 14:37:02.906112091 +0200
+++ pytorch-1.8.0-new/c10/core/TensorImpl.h	2021-10-07 14:37:14.542274541 +0200
@@ -1890,7 +1890,7 @@ protected:
 //    tensor type id
 //    miscellaneous bitfield
 //
-static_assert(sizeof(void*) != sizeof(int64_t) || // if 64-bit...
+static_assert(sizeof(void*) <= sizeof(int64_t) || // if 64-bit...
               sizeof(TensorImpl) == sizeof(int64_t) * 24,
               "You changed the size of TensorImpl on 64-bit arch."
               "See Note [TensorImpl size constraints] on how to proceed.");
