diff -rup a/pytorch/setup.py b/pytorch/setup.py
--- a/pytorch/setup.py	2020-01-24 10:52:00.235658726 +0100
+++ b/pytorch/setup.py	2020-01-24 10:52:22.382686744 +0100
@@ -293,7 +293,6 @@ def build_deps():
             sys.exit(1)
 
     check_file(os.path.join(third_party_path, "gloo", "CMakeLists.txt"))
-    check_file(os.path.join(third_party_path, "pybind11", "CMakeLists.txt"))
     check_file(os.path.join(third_party_path, 'cpuinfo', 'CMakeLists.txt'))
     check_file(os.path.join(third_party_path, 'tbb', 'Makefile'))
     check_file(os.path.join(third_party_path, 'onnx', 'CMakeLists.txt'))
@@ -846,8 +845,6 @@ if __name__ == '__main__':
                 'include/torch/csrc/jit/testing/*.h',
                 'include/torch/csrc/onnx/*.h',
                 'include/torch/csrc/utils/*.h',
-                'include/pybind11/*.h',
-                'include/pybind11/detail/*.h',
                 'include/TH/*.h*',
                 'include/TH/generic/*.h*',
                 'include/THC/*.cuh',
