diff --git a/pyNN/recording/files.py b/pyNN/recording/files.py
index 67c574ae..15404329 100644
--- a/pyNN/recording/files.py
+++ b/pyNN/recording/files.py
@@ -50,7 +50,7 @@ def savez(file, *args, **kwds):
     import zipfile
     from numpy.lib import format
 
-    if isinstance(file, basestring):
+    if isinstance(file, str):
         if not file.endswith('.npz'):
             file = file + '.npz'
 
@@ -61,7 +61,7 @@ def savez(file, *args, **kwds):
             raise ValueError("Cannot use un-named variables and keyword %s" % key)
         namedict[key] = val
 
-    zip = zipfile.ZipFile(file, mode="wb")
+    zip = zipfile.ZipFile(file, mode="w")
 
     # Place to write temporary .npy files
     #  before storing them in the zip. We need to path this to have a working
@@ -220,7 +220,7 @@ class NumpyBinaryFile(BaseFile):
     def write(self, data, metadata):
         __doc__ = BaseFile.write.__doc__
         self._check_open()
-        metadata_array = numpy.array(metadata.items(), dtype=(str, float))
+        metadata_array = numpy.array(list(metadata.items()), dtype=object)
         savez(self.fileobj, data=data, metadata=metadata_array)
 
     def read(self):
@@ -234,7 +234,7 @@ class NumpyBinaryFile(BaseFile):
         __doc__ = BaseFile.get_metadata.__doc__
         self._check_open()
         D = {}
-        for name, value in numpy.load(self.fileobj)['metadata']:
+        for name, value in numpy.load(self.fileobj, allow_pickle=True)['metadata']:
             try:
                 D[name] = eval(value)
             except Exception:
